File: CarLine.DataCleanUp/Models/CleanupResult.cs
namespace CarLine.DataCleanUp.Models;

public class CleanupResult
{
    public bool Success { get; set; }
    public string Message { get; set; } = string.Empty;
    public long RowsRead { get; set; }
    public long RowsWritten { get; set; }
    public long RowsDropped { get; set; }
    public int Errors { get; set; }
    public string? BlobName { get; set; }
    public TimeSpan Duration { get; set; }
}

File: CarLine.DataCleanUp.FSharp/Core/CleanupResult.fs
namespace CarLine.DataCleanUp.FSharp

open System

/// Immutable summary returned after a cleanup run completes.
type CleanupResult = {
    Success    : bool
    Message    : string
    RowsRead   : int64
    RowsWritten: int64
    RowsDropped: int64
    Errors     : int
    BlobName   : string option   // 'option' replaces nullable string?
    Duration   : TimeSpan
}

